source/common/http/http2/codec_impl.cc

```diff
@@ -1386,6 +1396,74 @@ ConnectionImpl::ClientHttp2Options::ClientHttp2Options(
       options_, ::Envoy::Http2::Utility::OptionsLimits::DEFAULT_MAX_CONCURRENT_STREAMS);
 }
 
+void ConnectionImpl::dumpState(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+  os << spaces << "Http2::ConnectionImpl " << this << DUMP_MEMBER(max_headers_kb_)
+     << DUMP_MEMBER(max_headers_count_) << DUMP_MEMBER(per_stream_buffer_limit_)
+     << DUMP_MEMBER(allow_metadata_) << DUMP_MEMBER(stream_error_on_invalid_http_messaging_)
+     << DUMP_MEMBER(is_outbound_flood_monitored_control_frame_)
+     << DUMP_MEMBER(skip_encoding_empty_trailers_) << DUMP_MEMBER(dispatching_)
+     << DUMP_MEMBER(raised_goaway_) << DUMP_MEMBER(pending_deferred_reset_) << '\n';
+
+  // Dump the protocol constraints
+  DUMP_DETAILS(&protocol_constraints_);
+
+  os << spaces << "Number of active streams: " << active_streams_.size() << " Active Streams:\n";
+  std::for_each_n(active_streams_.begin(), std::min<size_t>(active_streams_.size(), 100),
+                  [&](auto& stream) { DUMP_DETAILS(stream); });
+
+  // Dump the active slice
+  if (current_slice_ == nullptr) {
+    // No current slice, use macro for consistent formatting.
+    os << spaces << "current_slice_: null\n";
+  } else {
+    auto slice_view =
+        absl::string_view(static_cast<const char*>(current_slice_->mem_), current_slice_->len_);
+
+    os << spaces << "current slice length: " << slice_view.length() << " contents: \"";
+    StringUtil::escapeToOstream(os, slice_view);
+    os << "\"\n";
+  }
+}
+
+void ConnectionImpl::StreamImpl::dumpState(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+  os << "ConnectionImpl::StreamImpl " << this << DUMP_MEMBER(stream_id_)
```

Comment:
nit: I think you're missing spaces in this line

os << spaces << "ConnectionImpl::StreamImpl "

Reply:
Done.